### PR TITLE
some more build and test improvements

### LINKED ISSUE #27219: Errors when generating the Win docs on AppVeyor give zero information
```
Building HTML documentation.
/cygdrive/c/projects/julia/usr/bin/julia.exe --color=yes `cygpath -w /cygdrive/c/projects/julia/doc/make.jl` > docbuild.log 2>&1
make[2]: *** [Makefile:35: html] Error 1
make[2]: Leaving directory '/cygdrive/c/projects/julia/doc'
make[1]: *** [Makefile:88: docs] Error 2
make[1]: Leaving directory '/cygdrive/c/projects/julia'
make: *** [Makefile:50: /cygdrive/c/projects/julia/doc/_build/html/en/index.html] Error 2
Command exited with code 2
```

### PR DESCRIPTION
**Pull request by vtjnash**
fixes #27219

### CODE REVIEW

**Comment by KristofferC** [comment]
Fixes https://github.com/JuliaLang/julia/issues/27219 it seems.

**Comment by vtjnash** [comment]
Yes – I didn't realize we had a specific issue number for that

File: doc/Makefile [hunk 1]
```diff
@@ -35,13 +35,7 @@ cleanall: clean
 
 html: deps
 	@echo "Building HTML documentation."
-ifneq ($(OS),WINNT)
 	$(JULIA_EXECUTABLE) --color=yes $(call cygpath_w,$(SRCDIR)/make.jl) $(DOCUMENTER_OPTIONS)
-else
-# work around issue #11727, windows output redirection breaking on buildbot
```
Comment (ararslan) [inline doc/Makefile]: Is this not still a problem then?

Comment (vtjnash) [reply doc/Makefile]: No, #11727 is closed and fixed

**Pull request merged**